cluster/src/test/scala/org/apache/spark/sql/execution/benchmark/TAQTest.scala

```diff
@@ -81,6 +80,23 @@ class TAQTest extends SnappyFunSuite {
       tradeSize, numDays, queryNumber = 3, numIters, doInit = false)
   }
 
+  test("SNAP-1787"){
+    val tableName = "complexTypeTable"
+    val arr1 = Array(Decimal("4.92"), Decimal("51.98"))
+    val arr2 = Array(4.92f, 51.98f)
+    val arr3 = Array(4921212121L, 5198121212L)
+    val arr4 = Array(true, false)
+    val arr5 = Array(4.92, 51.98)
+    val data = scala.collection.mutable.ArrayBuffer[ComplexTypeData]()
+    data +=  new ComplexTypeData(arr1, arr2, arr3, arr4, arr5)
+    val rdd = snc.sparkContext.parallelize(data, 8)
+    val dataDF = snc.createDataFrame(rdd)
+    snc.createTable(s"$tableName", "column", dataDF.schema, Map.empty[String, String])
+    dataDF.write.insertInto(s"$tableName")
+    assert(snc.sql(s"select * from $tableName").count() == 1)
+    snc.dropTable(s"$tableName", true)
```

Comment:
This does not check for ComplexTypeSerializer? Check for its serialize/deserialize compatibility for Array[Decimal].